advanced/i2c-driver-exercise/src/imc42670p.rs

```diff
@@ -0,0 +1,59 @@
+#![deny(unsafe_code)]
+#![no_std]
+
+use embedded_hal::blocking::i2c;
+
+/// IMC42670P device driver.
+#[derive(Debug)]
+pub struct IMC42670P<I2C> {
+    /// The concrete I²C device implementation.
+    i2c: I2C,
+
+    /// Device address
+    address: SlaveAddr,
+}
+
+// see Table 3.3.2
+#[derive(Debug, Clone, Copy, PartialEq)]
+pub enum SlaveAddr {
+
+    AD0 = 0b110_1000,
+    AD1 = 0b110_1001,
+}
+
+impl<I2C, E>IMC42670P<I2C>
+where
+    I2C: i2c::WriteRead<Error = E> + i2c::Write<Error = E>,
+{
+    /// Add doc comments!
+    pub fn new(i2c: I2C, address: SlaveAddr) -> Result<Self, E> {
```

Comment:
We should explain here that if you own the I2C device, no-one else can use the bus whilst this driver exists. The users options are then either
* live with this, and create and destroy the driver every time they want to talk to the sensor,
* take the I2C bus as an argument to the `read_XXX` functions,
* use https://docs.rs/shared-bus/latest/shared_bus/.

Reply:
I have not made the skeletons yet, one reason is a priority issue and the second is that I wanted someone to look over the solutions I expect to not have to change the same code in multiple places. I added a readme to the advanced folder that explains the purpose of the files.